src/coreclr/debug/daccess/daccess.cpp

```diff
@@ -264,7 +264,7 @@ SplitFullName(_In_z_ PCWSTR fullName,
     else
     {
         *params = NULL;
-        memberEnd = fullName + (u16_strlen(fullName) - 1);
+        memberEnd = fullName + (minipal_u16_strlen((const CHAR16_T*)fullName) - 1);
```

Comment:
Why do we need these `(const CHAR16_T*)` casts? Should `PCWSTR` line up with `(const CHAR16_T*)`?

Reply:
It doesn't line up on Windows, as `wchar_t*` isn't castable to `unsigned short*` on Windows.

Reply:
Do you expect that we are going to reconcile this or is this here to stay?

If it is here to stay, I think we should keep `u16_strlen` and similar methods to avoid the casts everywhere and limit the use of the minipal API to places that do not need casts. I do not think we want to have these casts all over the place. I am sorry that I have not realized that replacing `u16_strlen` will result in these casts when I have made the suggestion to inline it.

Reply:
I can revert the `u16_strlen` changes. No worries!